sdk/core/core-lro/review/core-lro.api.md

```diff
@@ -10,53 +10,45 @@ import { AbortSignalLike } from '@azure/abort-controller';
 export type CancelOnProgress = () => void;
 
 // @public
-export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): Promise<SimplePollerLike<TState, TResult>>;
+export function createHttpPoller<TResult, TState extends OperationState<TResult>>(lro: LongRunningOperation, options?: CreateHttpPollerOptions<TResult, TState>): PollerLike<TState, TResult>;
 
 // @public
 export interface CreateHttpPollerOptions<TResult, TState> {
     intervalInMs?: number;
-    processResult?: (result: unknown, state: TState) => TResult;
+    processResult?: (result: unknown, state: TState) => TResult | Promise<TResult>;
     resolveOnUnsuccessful?: boolean;
-    resourceLocationConfig?: LroResourceLocationConfig;
+    resourceLocationConfig?: ResourceLocationConfig;
     restoreFrom?: string;
-    updateState?: (state: TState, response: LroResponse) => void;
+    updateState?: (state: TState, response: OperationResponse) => void;
     withOperationLocation?: (operationLocation: string) => void;
 }
 
+// @public
+export function deserializeState<TState>(serializedState: string): RestorableOperationState<TState>;
+
 // @public
 export interface LongRunningOperation<T = unknown> {
-    requestMethod?: string;
-    requestPath?: string;
-    sendInitialRequest: () => Promise<LroResponse<unknown>>;
+    sendInitialRequest: () => Promise<OperationResponse<unknown>>;
     sendPollRequest: (path: string, options?: {
         abortSignal?: AbortSignalLike;
-    }) => Promise<LroResponse<T>>;
+    }) => Promise<OperationResponse<T>>;
 }
 
 // @public
-export class LroEngine<TResult, TState extends PollOperationState<TResult>> extends Poller<TState, TResult> {
-    constructor(lro: LongRunningOperation<TResult>, options?: LroEngineOptions<TResult, TState>);
-    delay(): Promise<void>;
-}
-
-// @public
-export interface LroEngineOptions<TResult, TState> {
-    intervalInMs?: number;
-    isDone?: (lastResponse: unknown, state: TState) => boolean;
-    lroResourceLocationConfig?: LroResourceLocationConfig;
-    processResult?: (result: unknown, state: TState) => TResult;
-    resolveOnUnsuccessful?: boolean;
-    resumeFrom?: string;
-    updateState?: (state: TState, lastResponse: RawResponse) => void;
+export interface OperationConfig {
+    initialUri?: string;
```

Comment:
can we rename this to `initialUrl`?

Reply:
yes, I name it as initialUri because of .Net using that name. Previously we have a topic for cross language rehydration([gist](https://gist.github.com/live1206/bbff9c1e0fa46b99647a33386ca0a98d?permalink_comment_id=4853073#implementation-details)). I want to reduce some breakings between us. 

Seems this topic is delayed now and I think we could follow our own naming convention.